pkg/cfn/builder/nodegroup.go

```diff
@@ -242,7 +242,7 @@ func newLaunchTemplateData(n *NodeGroupResourceSet) *gfn.AWSEC2LaunchTemplate_La
 		ImageId:  gfn.NewString(n.spec.AMI),
 		UserData: n.userData,
 		NetworkInterfaces: []gfn.AWSEC2LaunchTemplate_NetworkInterface{{
-			AssociatePublicIpAddress: gfn.NewBoolean(!n.spec.PrivateNetworking),
+			AssociatePublicIpAddress: nil,
```

Comment:
this one is for new CF only, so can we just remove this line, unless it's used for updating CF as well ?

Reply:
This is so new stacks don't get this set. I think it has been set by default so now we have to explicitly unset it.

At least, the current functionality enables this and I couldn't find where that was done so I assumed that it is done by default.

Reply:
I think we should add a short comment stating that this is now set at the subnet level.